contracts/hooks/examples/LimitOrder.sol

```diff
@@ -144,27 +142,24 @@ contract LimitOrder is BaseHook {
         // order fills are the opposite of swap fills, hence the inversion below
         bool zeroForOne = !params.zeroForOne;
         for (; lower <= upper; lower += key.tickSpacing) {
-            _fillEpoch(key, lower, zeroForOne);
+            _fillEpoch(sender, key, lower, zeroForOne);
         }
 
         setTickLowerLast(key.toId(), tickLower);
         return LimitOrder.afterSwap.selector;
     }
 
-    function _fillEpoch(PoolKey calldata key, int24 lower, bool zeroForOne) internal {
+    function _fillEpoch(address sender, PoolKey calldata key, int24 lower, bool zeroForOne) internal {
         Epoch epoch = getEpoch(key, lower, zeroForOne);
         if (!epoch.equals(EPOCH_DEFAULT)) {
             EpochInfo storage epochInfo = epochInfos[epoch];
 
             epochInfo.filled = true;
 
-            (uint256 amount0, uint256 amount1) = abi.decode(
-                poolManager.lock(
-                    address(this),
-                    abi.encodeCall(this.lockAcquiredFill, (key, lower, -int256(uint256(epochInfo.liquidityTotal))))
-                ),
-                (uint256, uint256)
-            );
+            address locker = poolManager.getLocker();
+            require(locker == sender, "invalid locker");
```

Comment:
Is this the right way to remove nested locking?

Reply:
replacing the .lock with _lockAcquiredFill is right

I dont think we need this require statement since its an internal function thats only called from `afterSwap` which already has `poolManagerOnly` modifier